src/hdmf/common/table.py

```diff
@@ -596,23 +585,9 @@ def add_row(self, **kwargs):
         extra_columns = set(list(data.keys())) - set(list(self.__colids.keys()))
         missing_columns = set(list(self.__colids.keys())) - set(list(data.keys()))
 
-        bad_data = []
         for colname, colnum in self.__colids.items():
             if colname not in data:
                 raise ValueError("column '%s' missing" % colname)
-            col = self.__df_cols[colnum]
-            if isinstance(col, VectorIndex):
-                continue
-            else:
-                if col.term_set is not None:
-                    if col.term_set.validate(term=data[colname]):
-                        continue
-                    else:
-                        bad_data.append(data[colname])
-
-        if len(bad_data)!=0:
-            msg = ('"%s" is not in the term set.' % ', '.join([str(item) for item in bad_data]))
-            raise ValueError(msg)
```

Comment:
What is nice about having this validation code here is that all the values of the row are checked before any values in the row are added. If we leave VectorData.add_row to handle the validation, one value at a time, the user might end up in a situation where the value of the first column is valid and added and the value of the second column is invalid and raises an error. This will result in an invalid table that is difficult to fix. Can you add this validation code back plus a comment about why it's here.

Reply:
This was the double validation we talked about yesterday. I agree the value here is probably worth the double validation.